### PR TITLE
Ignore empty messages logging

### PR DESCRIPTION
**Pull request by mehlah**
This is basically a replacement for #12 which is stale for a couple years.
Empty messages (looking at you Rails!) are now ignored, to avoid raised errors by `Notifier#check_presence_of_mandatory_attributes`

PS: I signed and sent the CA.

### CODE REVIEW

File: lib/gelf/logger.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ module LoggerCompatibility
     attr_accessor :formatter
 
     # Use it like Logger#add... or better not to use at all.
-    def add(level, message = nil, progname = nil)
+    def add(level, message = nil, progname = nil, &block)
```
Comment (mehlah) [inline lib/gelf/logger.rb]: Although this can be optional, for code clarity at least I added it back.
It has been removed at https://github.com/Graylog2/gelf-rb/commit/1656370e3964280c8baad374fdbbce44a5dbbdaf#diff-9fe274efe6e812365edf617e12452b0dL8

File: test/test_logger.rb [hunk 2]
```diff
@@ -14,6 +14,16 @@ class TestLogger < Test::Unit::TestCase
...
     context "#add" do
+      # logger.add(Logger::INFO, nil)
+      should 'implement add method with level, message and facility from defaults' do
+        @logger.expects(:notify_with_level!).with do |level, hash|
+          level == GELF::INFO &&
+          hash['short_message'] == 'gelf-rb' &&
+          hash['facility'] == 'gelf-rb'
+        end
+        @logger.add(GELF::INFO, nil)
+      end
```
Comment (mehlah) [inline test/test_logger.rb]: I added this test to explicitly highlight this behavior.
The [original idea](https://github.com/Graylog2/gelf-rb/commit/5a2c852e97743c18221dccfae032b741b527e643) was to have an consistent API with `Logger#add`, but it wasn't really the case. This could be refined I guess in a separate PR.

**Comment by dennisoelkers** [comment]
Hey @mehlah, thanks for submitting the PR. Unfortunately we have not received your signed CLA. But in the meantime we have switched our process to CLA assistant, so all you need to do is click on "Details" next to the `licence/cla` check and accept our CLA.

**Comment by mehlah** [comment]
@dennisoelkers done 📝

File: lib/gelf/logger.rb (hunk 1 above)
Comment (milgner) [inline lib/gelf/logger.rb]: But what is the point of passing a block if it is never called? Is it required for compatibility?

Comment (mehlah) [inline lib/gelf/logger.rb]: It makes it clear from the `add` method signature that it takes a block and therefore will presumably do something with it. It's more a documentation convenience for readers than a compatibility issue (I'm not aware of any compatibility issue btw).

Let me know if this doesn't fit the project code style so I can remove it :)

Comment (mehlah) [inline lib/gelf/logger.rb]: Sorry for my late answer.
I'm not sure to get you well. Do you prefer to remove the `&block` param, or something else? Actually I just saw that Ruby's `Logger#add` removed it also at https://github.com/ruby/ruby/commit/460e9d8d83b22efad02f69925492195124f52a86.

Comment (milgner) [inline lib/gelf/logger.rb]: Yes, they removed it but since there may very well still be a few users of this library who are stuck on Ruby 1.9, we can keep the block argument for the moment. I'd just prefer to actually use it (like `message ||= block.call unless block.nil?`) so that it isn't a dead argument.

Comment (mehlah) [inline lib/gelf/logger.rb]: Okay. I did what you suggest at https://github.com/Graylog2/gelf-rb/pull/50/commits/913baae452c15dd45c2b5bdaf3a9bd38210e0cdb.

Actually I think adding back this argument wasn't a good idea from me. It have an impact on performance as having `&block` arg leads to instantiating a Proc object.

What do you think about removing it back (like it was at first place)? I'm not aware about any impact on Ruby 1.9.

Comment (mehlah) [inline lib/gelf/logger.rb]: 👋 @milgner, any feedback on the above?

**Comment by paulRbr** [comment]
This PR looks good. Any change it could be merged in?
